DSS/DSS.FuzzyInference/MeasurementHandler.cs

```diff
@@ -36,17 +47,38 @@ public void Handle(object obje)
                 //TODO: wight data is going to be wrapped inside of an object!
                 var val = float.Parse( obj.value_info);
 
-				var kg = new RmqAPI("").GetLatestWeightMeasurement();
+				var kg = API.GetLatestWeightMeasurement();
+
+               if (Math.Abs(val - kg) > 2)                {                     var msg = val > kg ? "Have lighter meals" : "Have more consistent meals";                     result.Add("Abnormal change in weight noticed - " + msg);                     API.PushJournalEntry(msg, "Abnormal change in weight noticed");                         					obj.ok = false;                     //and push notification                  }                 else                  {                     obj.ok = true;                     API.PushJournalEntry("Weight is OK", "Weight is OK");                }             }
+            else if(obj.measurement_type == "pulse") 
+            {
+				var val = float.Parse(obj.value_info);
+                var min = 50;
+                var max = 120;
+
+                if (val < min || val > max) 
+                {
+					obj.ok = false;
+
+					if(API.AreLastNHeartRateCritical(3, min, max)) 
+                    {
+                        //generate an event and push it to the event exchange 
+                    }
+
 
-                if (Math.Abs(val - kg) > 2)
-				{
-                        var msg = val > kg ? "Have lighter meals" : "Have more consistent meals";
-                        result.Add("Abnormal change in weight noticed - " + msg);
-						new RmqAPI("").PushJournalEntry(msg, "Abnormal change in weight noticed");
+					API.PushJournalEntry("Pulse is abnormal", "Pulse is abnormal");
 				}
+                else 
+                {
```

Comment:
You can remove the else branch, since the Journal must contain only alerts for **abnormal** values.
If the measurements are `ok`, then they should only be inserted in the Measurements endpoint.